app/src/main/java/com/example/seetha/seasonalclone/adapters/SeasonalAdapter.kt

```diff
@@ -1,33 +1,89 @@
 package com.example.seetha.seasonalclone.adapters
 
 import android.support.v7.widget.RecyclerView
-import android.view.View
 import android.view.ViewGroup
 import com.example.seetha.seasonalclone.models.ProduceItem
 import views.BrowseItemView
+import views.HeaderItemView
 
-class SeasonalAdapter(private var items: List<ProduceItem>) : RecyclerView.Adapter<SeasonalAdapter.ViewHolder>() {
+class SeasonalAdapter(private val items: List<ProduceItem>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    val FRUIT_TYPE = "Fruit"
+    val VEGETABLE_TYPE = "Vegetable"
+    val HERB_TYPE = "Herb"
+    val LEGUME_TYPE = "Legume"
+    val NUT_TYPE = "Nut"
+
+    private val itemsWithHeaders = sortItemsByHeader(items)
 
     init {
         notifyDataSetChanged()
     }
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-        val itemView = BrowseItemView.inflate(parent)
-        return ViewHolder(itemView)
+    private fun sortItemsByHeader(items: List<ProduceItem>): List<Any> {
+
+        val displayItems = items.filter { item -> item.display }
+        val sortedProduce = mutableListOf<Any>()
+        if (displayItems.isNotEmpty()) {
+            sortedProduce.add( "Fruits")
+            sortedProduce.addAll(displayItems.filter { displayItem -> displayItem.type == FRUIT_TYPE })
+            sortedProduce.add("Veggies")
+            sortedProduce.addAll(displayItems.filter { displayItem ->
```

Comment:
FYI, we'll still add "Fruits" even if we don't have any fruits, and "Veggies" even if we don't have any veggies. Might not be a big deal for this exercise since California data always has both in season for every month.

Reply:
Yes, I wanted to leave those in.